src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -406,15 +406,16 @@ class Address {
     : _base(r), _index(noreg), _offset(0), _mode(base_plus_offset), _target(0) { }
   Address(Register r, int o)
     : _base(r), _index(noreg), _offset(o), _mode(base_plus_offset), _target(0) { }
-  Address(Register r, long o)
+  Address(Register r, int64_t o)
     : _base(r), _index(noreg), _offset(o), _mode(base_plus_offset), _target(0) { }
-  Address(Register r, long long o)
+#ifdef __APPLE__
+  // macosx wants all the overloads
+  Address(Register r, uintptr_t o)
     : _base(r), _index(noreg), _offset(o), _mode(base_plus_offset), _target(0) { }
-  Address(Register r, unsigned int o)
+  Address(Register r, ptrdiff_t o)
     : _base(r), _index(noreg), _offset(o), _mode(base_plus_offset), _target(0) { }
-  Address(Register r, unsigned long o)
-    : _base(r), _index(noreg), _offset(o), _mode(base_plus_offset), _target(0) { }
-  Address(Register r, unsigned long long o)
+#endif
+  Address(Register r, uint64_t o)
     : _base(r), _index(noreg), _offset(o), _mode(base_plus_offset), _target(0) { }
```

Comment:
The above could be written (from an overlapping CS, with additional initialisation)
```suggestion
    Address(Register r) : Address(r, 0) {}
    Address(Register r, int32_t o) : Address(r, int64_t(o)) {}
    Address(Register r, int64_t o) : _mode(base_plus_offset),
      _base(r), _index(noreg), _offset(o), _extend(lsl(0)), _target(nullptr) {}

    Address(Register r, uint32_t o) : Address(r, uint64_t(o)) {}
    Address(Register r, uint64_t o) : _mode(base_plus_offset),
      _base(r), _index(noreg), _offset(o), _extend(lsl(0)), _target(nullptr) {}
```